internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/handler/ScanTask.java

```diff
@@ -0,0 +1,693 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc., WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.security.tools.scanmanager.scanners.veracode.handler;
+
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.SftpException;
+import com.veracode.apiwrapper.wrappers.ResultsAPIWrapper;
+import com.veracode.apiwrapper.wrappers.UploadAPIWrapper;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.log4j.Logger;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.wso2.security.tools.scanmanager.common.model.LogType;
+import org.wso2.security.tools.scanmanager.common.model.ScanStatus;
+import org.wso2.security.tools.scanmanager.scanners.common.ScannerConstants;
+import org.wso2.security.tools.scanmanager.scanners.common.config.YAMLConfigurationReader;
+import org.wso2.security.tools.scanmanager.scanners.common.exception.ScannerException;
+import org.wso2.security.tools.scanmanager.scanners.common.util.CallbackUtil;
+import org.wso2.security.tools.scanmanager.scanners.veracode.VeracodeScannerConstants;
+import org.wso2.security.tools.scanmanager.scanners.veracode.config.VeracodeScannerConfiguration;
+import org.wso2.security.tools.scanmanager.scanners.veracode.model.ScanContext;
+import org.wso2.security.tools.scanmanager.scanners.veracode.util.FileUtil;
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.concurrent.TimeUnit;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPathExpressionException;
+
+import static org.wso2.security.tools.scanmanager.common.model.LogType.ERROR;
+
+/**
+ * Represents the scan handling tasks.
+ */
+public class ScanTask implements Runnable {
+
+    private static final Logger log = Logger.getLogger(ScanTask.class);
+    // Veracode upload API wrapper.
+    private UploadAPIWrapper uploadAPIWrapper;
```

Comment:
Wrappers should not be class level fields. Please check the comment added in VeracodeScanner.